lib/speech/index.js

```diff
@@ -0,0 +1,903 @@
+ * @param {number} syncRecognizeRequest.config.sampleRate - Sample rate in Hertz
+ *     of the audio data sent in all RecognitionAudio messages. See
+ *     [`RecognitionConfig.sampleRate`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ * @param {string=} syncRecognizeRequest.config.languageCode - See
+ *     [`RecognitionConfig.languageCode`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ *     Default: `"en-US"`.
+ * @param {number=} syncRecognizeRequest.config.maxAlternatives - See
+ *     [`RecognitionConfig.maxAlternatives`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ *     Default: `1`.
+ * @param {boolean=} syncRecognizeRequest.config.profanityFilter - See
+ *     [`RecognitionConfig.profanityFilter`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ *     Default: `false`.
+ * @param {string[]=} syncRecognizeRequest.config.speechContext - See
+ *     [`RecognitionConfig.speechContext`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ *     Default: `[]`.
+ * @param {object} syncRecognizeRequest.audio - A `RecognitionAudio` object.
+ *     See [`RecognitionAudio`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionAudio).
+ * @param {buffer} syncRecognizeRequest.audio.content - The audio data bytes
+ *     encoded as specified in `RecognitionConfig`.
+ * @param {string} syncRecognizeRequest.audio.uri - URI that points to a file
+ *     that contains audio data bytes as specified in `RecognitionConfig`.
+ *     Currently, only Google Cloud Storage URIs are supported, which must be
+ *     specified in the following format: `gs://bucket_name/object_name`.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {object[]} callback.results - Array of speech recognition results. See
+ *     [`SpeechRecognitionResult`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.SpeechRecognitionResult).
+ * @param {object} callback.apiResponse - Raw API response. See
+ *     [`SyncRecognizeResponse`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#syncrecognizeresponse).
+ *
+ * @example
+ * //-
+ * // <h3>Send raw audio</h3>
+ * //-
+ * var fs = require('fs');
+ *
+ * fs.readFile('./system-test/data/bridge.raw', function (err, rawAudio) {
+ *   // See the link in the parameters for `SyncRecognizeRequest`.
+ *   var syncRecognizeRequest = {
```

Comment:
I purposefully kept `syncRecognize`, `asyncRecognize`, and `streamingRecognize` as lower-level 1:1 mappings to the grpc methods (following the example of gcloud-node's Vision API implementation, which has the low-level `annotate` method that doesn't add any sugar, but adds some nice `detect` methods that are easier to work with), without altering the interface defined in the proto. `syncDetect`, `asyncDetect`, and `streamingDetect` are the simplified methods that provide an easier interface to work with, like what you suggested in your comment.

> I'm assuming that the user can specify either raw audio content or a storage file path, but never both?

Correct.